ateam_kenobi/src/in_play_eval.cpp

```diff
@@ -0,0 +1,142 @@
+  if (game_command == ateam_common::GameCommand::ForceStart) {
+    in_play_ = true;
+  } else if (IsGameStopped(world)) {
+    in_play_ = false;
+    ball_start_pos_.reset();
+  } else if (IsStopCommandEnding(world)) {
+    SetTimeout(world);
+    SetDistanceThreshold(world);
+  } else if (IsGameResuming(world)) {
+    ball_start_pos_ = world.ball.pos;
+    timeout_start_ = std::chrono::steady_clock::now();
+  } else if (HasTimeoutExpired()) {
+    in_play_ = true;
+  } else if (HasBallMoved(world)) {
+    in_play_ = true;
+  }
+
+  world.in_play = in_play_;
+  prev_game_command_ = game_command;
+}
+
+void InPlayEval::SetTimeout(const World & world)
+{
+  switch (world.referee_info.running_command) {
+    case ateam_common::GameCommand::PrepareKickoffOurs:
+    case ateam_common::GameCommand::PrepareKickoffTheirs:
+      timeout_duration_ = std::chrono::seconds(10);
+      break;
+    case ateam_common::GameCommand::DirectFreeOurs:
+    case ateam_common::GameCommand::DirectFreeTheirs:
+      // Division A
+      // timeout_duration_ = std::chrono::seconds(5);
+      // Division B
+      timeout_duration_ = std::chrono::seconds(10);
+      break;
+    default:
+      timeout_duration_.reset();
+      break;
+  }
+}
+
+void InPlayEval::SetDistanceThreshold(const World & world)
+{
+  switch (world.referee_info.running_command) {
+    case ateam_common::GameCommand::PrepareKickoffOurs:
+    case ateam_common::GameCommand::PrepareKickoffTheirs:
+    case ateam_common::GameCommand::DirectFreeOurs:
+    case ateam_common::GameCommand::DirectFreeTheirs:
+    case ateam_common::GameCommand::PreparePenaltyOurs:
+    case ateam_common::GameCommand::PreparePenaltyTheirs:
+      ball_moved_threshold_ = 0.05;
+      break;
+    default:
+      ball_moved_threshold_ = std::numeric_limits<double>::infinity();
+      break;
+  }
+}
+
+
+bool InPlayEval::IsGameStopped(const World & world)
+{
+  return world.referee_info.running_command == ateam_common::GameCommand::Stop ||
+         world.referee_info.running_command == ateam_common::GameCommand::Halt;
+}
+
+bool InPlayEval::IsStopCommandEnding(const World & world)
+{
+  return world.referee_info.running_command != prev_game_command_ &&
+         prev_game_command_ == ateam_common::GameCommand::Stop;
+}
+
+
+bool InPlayEval::IsGameResuming(const World & world)
+{
+  if (world.referee_info.running_command == prev_game_command_) {
+    return false;
+  }
+
+  if (world.referee_info.running_command == ateam_common::GameCommand::NormalStart ||
+    world.referee_info.running_command == ateam_common::GameCommand::DirectFreeOurs ||
+    world.referee_info.running_command == ateam_common::GameCommand::DirectFreeTheirs)
+  {
+    return true;
+  }
+
+  return false;
+}
+
+bool InPlayEval::HasBallMoved(const World & world)
+{
+  if (!ball_start_pos_) {
+    return false;
+  }
+
+  return CGAL::approximate_sqrt(
+    CGAL::squared_distance(
+      world.ball.pos,
+      ball_start_pos_.value())) > ball_moved_threshold_;
```

Comment:
Do we want any epsilon here for the allowed ball distance?

Reply:
eh 🤷 it's probably fine to just use the distance from the rules